fix TestAddonSearchView intermittent

```
____ TestAddonSearchView.test_prevent_too_complex_to_determinize_exception _____

src/olympia/addons/tests/test_views.py:1770: in test_prevent_too_complex_to_determinize_exception

    data = self.perform_search(self.url, {'q': query})

src/olympia/addons/tests/test_views.py:1021: in perform_search

    response = self.client.get(url, data, **headers)

.tox/es/lib/python3.6/site-packages/django/test/testcases.py:85: in __exit__

    '%d. %s' % (i, query['sql']) for i, query in enumerate(self.captured_queries, start=1)

E   AssertionError: 1 != 0 : 1 queries executed, 0 expected

E   Captured queries were:

E   1. SELECT `waffle_switch`.`id`, `waffle_switch`.`name`, `waffle_switch`.`active`, `waffle_switch`.`note`, `waffle_switch`.`created`, `waffle_switch`.`modified` FROM `waffle_switch` WHERE `waffle_switch`.`name` = 'api-recommendations-priority'
```

--- a/src/olympia/addons/tests/test_views.py
+++ b/src/olympia/addons/tests/test_views.py
@@ -13,7 +13,6 @@ from elasticsearch import Elasticsearch
 from unittest.mock import patch
 from rest_framework.test import APIRequestFactory
 from waffle import switch_is_active
-from waffle.testutils import override_switch
 
 from olympia import amo
 from olympia.addons.models import Addon, AddonUser, Category, ReplacementAddon
@@ -967,8 +966,6 @@ class TestAddonSearchView(ESTestCase):
         self.url = reverse_ns('addon-search')
         self.create_switch('return-to-amo', active=True)
         switch_is_active('return-to-amo')
-        self.create_switch('api-recommendations-priority', active=True)
-        switch_is_active('api-recommendations-priority')
 
     def tearDown(self):
         super(TestAddonSearchView, self).tearDown()
@@ -1016,7 +1013,6 @@ class TestAddonSearchView(ESTestCase):
 
     def perform_search(self, url, data=None, expected_status=200,
                        expected_queries=0, **headers):
-        switch_is_active('api-recommendations-priority')  # just to cache it
         with self.assertNumQueries(expected_queries):
             response = self.client.get(url, data, **headers)
         assert response.status_code == expected_status, response.content
@@ -1766,7 +1762,6 @@ class TestAddonSearchView(ESTestCase):
         self.refresh()
         query = (u'남포역립카페추천 ˇjjtat닷컴ˇ ≡제이제이♠♣ 남포역스파 '
                  u'남포역op남포역유흥≡남포역안마남포역오피 ♠♣')
-        assert switch_is_active('api-recommendations-priority')
         data = self.perform_search(self.url, {'q': query})
         # No results, but no 500 either.
         assert data['count'] == 0
@@ -1800,14 +1795,6 @@ class TestAddonSearchView(ESTestCase):
         # addon2 and addon4 will be first because they're recommended
         assert ids == [addon2.id, addon4.id, addon1.id, addon3.id, addon5.id]
 
-        # But if the waffle is off the recommended add-ons don't have priority.
-        with override_switch('api-recommendations-priority', active=False):
-            assert not switch_is_active('api-recommendations-priority')
-            data = self.perform_search(self.url)  # No query.
-            ids = [result['id'] for result in data['results']]
-            assert ids == [
-                addon1.id, addon2.id, addon3.id, addon4.id, addon5.id]
-
 
 class TestAddonAutoCompleteSearchView(ESTestCase):
     client_class = APITestClient
@@ -1816,8 +1803,6 @@ class TestAddonAutoCompleteSearchView(ESTestCase):
 
     def setUp(self):
         super(TestAddonAutoCompleteSearchView, self).setUp()
-        self.create_switch('api-recommendations-priority', active=True)
-        switch_is_active('api-recommendations-priority')
         self.url = reverse_ns('addon-autocomplete', api_version='v5')
 
     def tearDown(self):
@@ -1941,7 +1926,6 @@ class TestAddonAutoCompleteSearchView(ESTestCase):
         self.refresh()
 
         # page_size should be ignored, we should get 10 results.
-        switch_is_active('api-recommendations-priority')  # just to cache it
         data = self.perform_search(self.url, {'page_size': 1})
         assert 'count' not in data
         assert 'next' not in data

--- a/src/olympia/search/filters.py
+++ b/src/olympia/search/filters.py
@@ -698,7 +698,8 @@ class SearchQueryFilter(BaseFilterBackend):
         secondary_should = self.secondary_should_rules(search_query, analyzer)
 
         # We alter scoring depending on add-on popularity and whether the
-        # add-on is reviewed & public & non-experimental.
+        # add-on is reviewed & public & non-experimental, and whether or not
+        # it's recommended.
         functions = [
             query.SF(
                 'field_value_factor',
@@ -713,15 +714,13 @@ class SearchQueryFilter(BaseFilterBackend):
                     Q('term', is_disabled=False)
                 )
             }),
+            query.SF({
+                'weight': 5.0,
+                'filter': (
+                    Q('term', is_recommended=True)
+                )
+            }),
         ]
-        if switch_is_active('api-recommendations-priority'):
-            functions.append(
-                query.SF({
-                    'weight': 5.0,
-                    'filter': (
-                        Q('term', is_recommended=True)
-                    )
-                }))
 
         # Assemble everything together
         qs = qs.query(
@@ -879,12 +878,10 @@ class SortingFilter(BaseFilterBackend):
         if not split_sort_params:
             # The default sort depends on the presence of a query: we sort by
             # relevance if we have a query, otherwise by recommended,downloads.
-            recommended_waffle_on = switch_is_active(
-                'api-recommendations-priority')
             split_sort_params = (
                 ['relevance'] if search_query_param else
-                ['recommended', 'users'] if recommended_waffle_on else
-                ['downloads'])
+                ['recommended', 'users']
+            )
 
         try:
             order_by = [self.SORTING_PARAMS[name] for name in

--- a/src/olympia/search/tests/test_filters.py
+++ b/src/olympia/search/tests/test_filters.py
@@ -8,7 +8,6 @@ from django.utils import translation
 
 from elasticsearch_dsl import Search
 from rest_framework import serializers
-from waffle.testutils import override_switch
 
 from olympia import amo
 from olympia.amo.tests import TestCase
@@ -127,13 +126,11 @@ class TestQueryFilter(FilterTestsBase):
             'weight': 5.0}
         return qs
 
-    @override_switch('api-recommendations-priority', active=True)
     def test_no_rescore_if_not_sorting_by_relevance(self):
         qs = self._test_q(
             self._filter(data={'q': 'tea pot', 'sort': 'rating'}))
         assert 'rescore' not in qs
 
-    @override_switch('api-recommendations-priority', active=True)
     def test_q(self):
         qs = self._test_q(self._filter(data={'q': 'tea pot'}))
 
@@ -176,12 +173,10 @@ class TestQueryFilter(FilterTestsBase):
             }
         }
 
-    @override_switch('api-recommendations-priority', active=True)
     def test_q_too_long(self):
         with self.assertRaises(serializers.ValidationError):
             self._filter(data={'q': 'a' * 101})
 
-    @override_switch('api-recommendations-priority', active=True)
     def test_fuzzy_single_word(self):
         qs = self._filter(data={'q': 'blah'})
         should = qs['query']['function_score']['query']['bool']['should']
@@ -213,7 +208,6 @@ class TestQueryFilter(FilterTestsBase):
         }
         assert expected in should
 
-    @override_switch('api-recommendations-priority', active=True)
     def test_fuzzy_multi_word(self):
         qs = self._filter(data={'q': 'search terms'})
         should = qs['query']['function_score']['query']['bool']['should']
@@ -245,7 +239,6 @@ class TestQueryFilter(FilterTestsBase):
         }
         assert expected in should
 
-    @override_switch('api-recommendations-priority', active=True)
     def test_no_fuzzy_if_query_too_long(self):
         def do_test():
             qs = self._filter(data={'q': 'this search query is too long.'})
@@ -289,7 +282,6 @@ class TestQueryFilter(FilterTestsBase):
             should = do_test()
             assert expected in should
 
-    @override_switch('api-recommendations-priority', active=True)
     def test_q_exact(self):
         qs = self._filter(data={'q': 'Adblock Plus'})
         should = qs['query']['function_score']['query']['bool']['should']
@@ -325,31 +317,6 @@ class TestQueryFilter(FilterTestsBase):
 
         assert expected in should
 
-    @override_switch('api-recommendations-priority', active=False)
-    def test_recommendations_waffle_off(self):
-        qs = self._filter(data={'q': 'tea pot'})
-
-        functions = qs['query']['function_score']['functions']
-        assert len(functions) == 2
-        assert functions[0] == {
-            'field_value_factor': {
-                'field': 'average_daily_users', 'modifier': 'log2p'
-            }
-        }
-        assert functions[1] == {
-            'filter': {
-                'bool': {
-                    'must': [
-                        {'term': {'is_experimental': False}},
-                        {'terms': {'status': (4,)}},
-                        {'exists': {'field': 'current_version'}},
-                        {'term': {'is_disabled': False}}
-                    ]
-                }
-            },
-            'weight': 4.0
-        }
-
 
 class TestReviewedContentFilter(FilterTestsBase):
 
@@ -375,16 +342,6 @@ class TestSortingFilter(FilterTestsBase):
         # queryset object.
         return {key[1:]: {'order': 'desc'}} if key.startswith('-') else key
 
-    @override_switch('api-recommendations-priority', active=False)
-    def test_sort_default_recommendations_waffle_off(self):
-        qs = self._filter(data={'q': 'something'})
-        assert qs['sort'] == [self._reformat_order('_score')]
-
-        qs = self._filter()
-        assert qs['sort'] == [
-            self._reformat_order('-weekly_downloads')]
-
-    @override_switch('api-recommendations-priority', active=True)
     def test_sort_default(self):
         qs = self._filter(data={'q': 'something'})
         assert qs['sort'] == [self._reformat_order('_score')]
@@ -402,9 +359,14 @@ class TestSortingFilter(FilterTestsBase):
         for param, es in SORTING_PARAMS.items():
             qs = self._filter(data={'sort': param})
             if param == 'relevance':
-                # relevance without q is ignored so default downloads is used
-                es = SORTING_PARAMS['downloads']
-            assert qs['sort'] == [self._reformat_order(es)]
+                # relevance without q is ignored so default sort is used
+                expected = [
+                    self._reformat_order('-is_recommended'),
+                    self._reformat_order('-average_daily_users')
+                ]
+            else:
+                expected = [self._reformat_order(es)]
+            assert qs['sort'] == expected
         # Having a search query does not change anything, the requested sort
         # takes precedence.
         for param in SORTING_PARAMS:
@@ -486,7 +448,6 @@ class TestSortingFilter(FilterTestsBase):
             {'random_score': {}}
         ]
 
-    @override_switch('api-recommendations-priority', active=True)
     def test_sort_recommended_only(self):
         # If you try to sort by just recommended it gets ignored
         qs = self._filter(data={'q': 'something', 'sort': 'recommended'})
@@ -497,7 +458,6 @@ class TestSortingFilter(FilterTestsBase):
             self._reformat_order('-is_recommended'),
             self._reformat_order('-average_daily_users')]
 
-    @override_switch('api-recommendations-priority', active=True)
     def test_sort_recommended_and_relevance(self):
         # with a q, recommended with relevance sort, recommended is ignored.
         qs = self._filter(

--- a/src/olympia/search/tests/test_search_ranking.py
+++ b/src/olympia/search/tests/test_search_ranking.py
@@ -4,8 +4,6 @@ import json
 from django.conf import settings
 from django.utils.encoding import force_text
 
-from waffle.testutils import override_switch
-
 from olympia import amo
 from olympia.amo.tests import (
     APITestClient, ESTestCase, reverse_ns, create_switch)
@@ -525,16 +523,16 @@ class TestRankingScenarios(ESTestCase):
 
     def test_scenario_tabby_cat(self):
         self._check_scenario('Tabby cat', (
-            ['Tabby Cat', 56.777256],
+            ['Tabby Cat', 283.8863],
         ))
 
     def test_scenario_tabbycat(self):
         self._check_scenario('tabbycat', (
-            ['Tabby Cat', 0.95445126],
-            ['OneTab', 0.8880905],
+            ['Tabby Cat', 4.7722564],
+            ['OneTab', 0.88809050],
             ['FoxyTab', 0.76142323],
             ['Authenticator', 0.68661183],
-            ['Tab Mix Plus', 0.51704407],
+            ['Tab Mix Plus', 0.517044070],
             ['Open Bookmarks in New Tab', 0.40527225],
             ['Tab Center Redux', 0.39011657],
             ['Open image in a new tab', 0.3115263],
@@ -543,8 +541,8 @@ class TestRankingScenarios(ESTestCase):
 
     def test_scenario_tabbbycat(self):
         self._check_scenario('tabbbycat', (
-            ['OneTab', 0.88739204],
-            ['Tabby Cat', 0.8728613],
+            ['Tabby Cat', 4.364307],
+            ['OneTab', 0.887392040],
             ['FoxyTab', 0.7608244],
             ['Authenticator', 0.6860718],
             ['Tab Mix Plus', 0.51663744],
@@ -556,10 +554,10 @@ class TestRankingScenarios(ESTestCase):
 
     def test_scenario_tabbicat(self):
         self._check_scenario('tabbicat', (
+            ['Tabby Cat', 3.4082708],
             ['OneTab', 0.8880905],
             ['FoxyTab', 0.76142323],
             ['Authenticator', 0.68661183],
-            ['Tabby Cat', 0.68165416],
             ['Tab Mix Plus', 0.51704407],
             ['Open Bookmarks in New Tab', 0.40527225],
             ['Tab Center Redux', 0.39011657],
@@ -794,56 +792,7 @@ class TestRankingScenarios(ESTestCase):
             [u'Foobar unique english', 4.957241],
         ), lang='en-US')
 
-    @override_switch('api-recommendations-priority', active=True)
-    def test_scenario_tabby_cat_recommendations_priority(self):
-        self._check_scenario('Tabby cat', (
-            ['Tabby Cat', 283.8863],
-        ))
-
-    @override_switch('api-recommendations-priority', active=True)
-    def test_scenario_tabbycat_recommendations_priority(self):
-        self._check_scenario('tabbycat', (
-            ['Tabby Cat', 4.7722564],
-            ['OneTab', 0.88809050],
-            ['FoxyTab', 0.76142323],
-            ['Authenticator', 0.68661183],
-            ['Tab Mix Plus', 0.517044070],
-            ['Open Bookmarks in New Tab', 0.40527225],
-            ['Tab Center Redux', 0.39011657],
-            ['Open image in a new tab', 0.3115263],
-            ['Open Image in New Tab', 0.24481377],
-        ))
-
-    @override_switch('api-recommendations-priority', active=True)
-    def test_scenario_tabbbycat_recommendations_priority(self):
-        self._check_scenario('tabbbycat', (
-            ['Tabby Cat', 4.364307],
-            ['OneTab', 0.887392040],
-            ['FoxyTab', 0.7608244],
-            ['Authenticator', 0.6860718],
-            ['Tab Mix Plus', 0.51663744],
-            ['Open Bookmarks in New Tab', 0.4049535],
-            ['Tab Center Redux', 0.38980976],
-            ['Open image in a new tab', 0.3112813],
-            ['Open Image in New Tab', 0.24462123],
-        ))
-
-    @override_switch('api-recommendations-priority', active=True)
-    def test_scenario_tabbicat_recommendations_priority(self):
-        self._check_scenario('tabbicat', (
-            ['Tabby Cat', 3.4082708],
-            ['OneTab', 0.8880905],
-            ['FoxyTab', 0.76142323],
-            ['Authenticator', 0.68661183],
-            ['Tab Mix Plus', 0.51704407],
-            ['Open Bookmarks in New Tab', 0.40527225],
-            ['Tab Center Redux', 0.39011657],
-            ['Open image in a new tab', 0.3115263],
-            ['Open Image in New Tab', 0.24481377],
-        ))
-
-    @override_switch('api-recommendations-priority', active=True)
-    def test_scenario_tab_recommendations_priority(self):
+    def test_scenario_tab(self):
         self._check_scenario('tab', (
             ['Tabby Cat', 9.2712755],
             ['OneTab', 4.11413670],
@@ -853,10 +802,4 @@ class TestRankingScenarios(ESTestCase):
             ['Open Bookmarks in New Tab', 2.9872396],
             ['Open image in a new tab', 2.5463212],
             ['Open Image in New Tab', 2.08934],
-        ))
-
-    @override_switch('api-recommendations-priority', active=True)
-    def test_scenario_websocket_recommendations_priority(self):
-        self._check_scenario('websocket', (
-            ['Simple WebSocket Client', 4.808697],
         ))

